### ISSUE TITLE
Interrupting queries through jobs table doesn't raise error

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4762]
Run this query:
```js
r.js("while (true) {}", {timeout: 600})
```

While it's running, run
```js
r.db('rethinkdb').table('jobs').delete()
```
a couple of times (because sometimes it kills itself before it gets to the `r.js` query).

Expected result: the `r.js` query should return an error
Actual result: the `r.js` query returns an empty non-error result

**Comment by danielmewes** [comment]
@Tryneus, @VeXocide any ideas?

**Comment by Tryneus** [comment]
@vexocide tracked this down to an `if` statement in `query_cache.cc`, which is mostly my code, so I'm going to take it from here.

**Comment by Tryneus** [comment]
Fix for this is up in review 3189.

**Comment by Tryneus** [comment]
The fix has been approved and merged to `next` in commit b74d398, and cherry-picked into `v2.1.x` in commit 7e07524.  Will be in release 2.1.3.
